pkg/controller/nodeipam/ipam/cidrset/cidr_set.go

```diff
@@ -263,3 +275,161 @@ func (s *CidrSet) getIndexForIP(ip net.IP) (int, error) {
 
 	return 0, fmt.Errorf("invalid IP: %v", ip)
 }
+
+// Map manages a set of CIDR ranges from which blocks of IPs can
+// be allocated from.
+type Map struct {
+	sync.Mutex
+	clusterCIDR   *net.IPNet
+	nodeMask      net.IPMask
+	maxCIDRs      uint64
+	nextCandidate uint64
+	used          map[uint64]bool
```

Comment:
`map[uint64]struct{}` makes the map values be of size 0:
https://gist.github.com/davecheney/3be245c92b61e5045f75#gistcomment-2671632

but `map[uint64]bool` is more readable.

Reply:
If we have a use for it you could probably get `sets.UInt64` added to apimachinery. There's already `sets.Int64` (but I think trying to use that would be uglier than trying to use `map[uint64]struct{}` directly). You could add a copy here for now and then we can eventually migrate to a version in apimachinery.

Reply:
wow, I didn't know what you meant before with the sets, my apologies ... I didn't know this `Sets` exist 😄